packages/thirdweb/src/react/core/hooks/transaction/usePrepareTransaction.ts

```diff
@@ -0,0 +1,201 @@
+
+type PickedQueryOptions = Prettify<
+  Pick<UseQueryOptions, "enabled"> & {
+    refetchInterval?: number;
+    retry?: number;
+  }
+>;
+
+/**
+ * A hook to prepare a transaction to be sent at a later time.
+ * @param extension - An extension to prepare.
+ * @param options - The extension params.
+ * @returns a query object.
+ * @example
+ * ```jsx
+ * import { usePrepareTransaction, useSendTransaction } from "thirdweb/react";
+ * import { mintTo } form "thirdweb/extensions/erc20";
+ *
+ * const { data: preparedTransaction, isLoading } = usePrepareTransaction(mintTo, { contract, to: address amount: "100" });
+ * const { mutate: sendTx, data: transactionResult } = useSendTransaction();
+ *
+ * // later send the transaction
+ * sendTx(preparedTransaction);
+ * ```
+ */
+export function usePrepareTransaction<
+  const abi extends Abi,
+  const params extends object,
+>(
+  extension: (
+    options: BaseTransactionOptions<params, abi>,
+  ) => PreparedTransaction<abi>,
+  options: BaseTransactionOptions<params, abi> & {
+    from?: Address;
+    queryOptions?: PickedQueryOptions;
+  },
+): UseQueryResult<Awaited<ReturnType<typeof toSerializableTransaction>>>;
+
+/**
+ * A hook to prepare a transaction to be sent at a later time.
+ * @param contractCallOptions - The contract call params
+ * @param options - options.
+ * @returns a query object.
+ * @example
+ * ```jsx
+ * import { usePrepareTransaction, useSendTransaction } from "thirdweb/react";
+ *
+ * const { data: preparedTransaction, isLoading } = usePrepareTransaction({
+ *  contract,
+ *  method: "function transfer(address to, uint256 value)",
+ *  params: [to, value],
+ * });
+ *
+ * const { mutate: sendTx, data: transactionResult } = useSendTransaction();
+ *
+ * // later send the transaction
+ * sendTx(preparedTransaction);
+ * ```
+ */
+export function usePrepareTransaction<const abi extends Abi>(
+  // biome-ignore lint/suspicious/noExplicitAny: too complicated to type sorry
+  contractCallOptions: PrepareContractCallOptions<abi, any>,
+  options?: {
+    from?: Address;
+    queryOptions?: PickedQueryOptions;
+  },
+): UseQueryResult<Awaited<ReturnType<typeof toSerializableTransaction>>>;
+
+/**
+ * A hook to prepare a transaction to be sent at a later time.
+ * @param txOptions - The contract call params
+ * @param options - options.
+ * @returns a query object.
+ * @example
+ * ```jsx
+ * import { usePrepareTransaction, useSendTransaction } from "thirdweb/react";
+ *
+ * const { data: preparedTransaction, isLoading } = usePrepareTransaction({
+ *  to: "0x1234567890123456789012345678901234567890",
+ *  chain: ethereum,
+ *  client: thirdwebClient,
+ *  value: toWei("1.0"),
+ *  gasPrice: 30n
+ * });
+ *
+ * const { mutate: sendTx, data: transactionResult } = useSendTransaction();
+ *
+ * // later send the transaction
+ * sendTx(preparedTransaction);
+ * ```
+ */
+export function usePrepareTransaction(
+  txOptions: PrepareTransactionOptions,
+  options?: {
+    from?: Address;
```

Comment:
i wonder if we should default this to the active account to match the sendTx behavior 🤔